pkg/destroy/aws/aws.go

```diff
@@ -1994,3 +2058,54 @@ func removeSharedTag(ctx context.Context, tagClients []*resourcegroupstaggingapi
 
 	return nil
 }
+
+// destroyedResources stores the results of attempts to destroy resources. The key is the ARN of the resource. The value
+// is whether the resource was successfully destroyed.
+type destroyedResources map[string]bool
+
+// record records where a resource was successfully destroyed. If the specified error is nil, then the resource was
+// destroyed successfully.
+func (r destroyedResources) record(arn string, err error) {
+	r[arn] = (err == nil)
+}
+
+// merge merges the record of destroyed resources from changes into r. If changes and r have results for the same
+// resource, the results from changes have precedence.
```

Comment:
This precedence doesn't make sense to me.  Wouldn't we want "if either value was `true`, there merged value is `true`"?  Because if Alice deleted A and merges with Bob who failed to delete A, A is still gone, regardless of Bob's failure.

Reply:
Or, thinking of it another way, if Alice failed to delete A even though Bob previously succeeded in deleting A, then A is still there regardless of whether Bob originally thought it had been deleted successfully.

Reply:
> ...regardless of whether Bob originally thought it had been deleted successfully.

That is very bad.  Why would we be ok with Bob thinking he deleted something when it had not been deleted?  I don't want to rely on Alice to come along and double-check him.

Reply:
OK. I don't care too much either way, so I will change it.